client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx

```diff
@@ -214,36 +193,41 @@ class RemoveDomainDialog extends Component {
 			return;
 		}
 		this.props.closeDialog();
-		this.setState( { step: 1 } );
+		this.setState( { domainValidated: false, step: 1 } );
 	};
 
 	render() {
 		const { purchase, translate, chatButton } = this.props;
 		const productName = getName( purchase );
+
+		// To be removed once "Never mind" is translated; the typo has been fixed in existing translations.
+		const closeDialogString = hasTranslation( 'Nevermind' )
+			? translate( 'Nevermind' )
+			: translate( 'Never mind' );
+
 		const buttons = [
 			{
 				action: 'cancel',
 				disabled: this.props.isRemoving,
-				label: this.state.step === 3 ? translate( 'Nevermind' ) : translate( 'Cancel' ),
-			},
-			{
-				action: 'remove',
-				additionalClassNames: [
-					this.props.isRemoving || this.state.isCheckingEmail ? 'is-busy' : '',
-					'dialog__button--domains-remove',
-				],
-				isPrimary: true,
-				label:
-					this.state.step === 3 ? (
-						<>
-							<Icon icon={ trash } size={ 18 } />
-							{ translate( 'Delete this domain' ) }
-						</>
-					) : (
-						translate( 'Continue' )
-					),
-				onClick: this.nextStep,
+				label: this.state.step === 3 ? closeDialogString : translate( 'Cancel' ),
 			},
+			...( this.state.step !== 2
+				? [
+						{
+							action: 'remove',
+							additionalClassNames: [
+								this.props.isRemoving || this.state.isCheckingEmail ? 'is-busy' : '',
```

Comment:
I know this issue already existed before your PR but I think that `isCheckingEmail` is not something in the state. Perhaps it was removed at some point and this was left behind? In any case it could be removed if you feel like it.

Reply:
Ah, nice spot! It is still used here:

https://github.com/Automattic/wp-calypso/blob/0ddcca0a061ae9c1d1f8acf477092ef4a5611a5a/client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx#L155-L169

But looking at this again, I'm not sure why we need to make a separate API call here. We already have the email data in state because this dialog is only used in `/me`, and that section doesn't finish loading until that data is loaded...so I think it's possible to just use `getCurrentUserEmail`, and not need to wait for another API call here. 

Let me know if I'm missing something obvious with those changes though!